proposals/secrets.md

```diff
@@ -0,0 +1,444 @@
+apiVersion: 'v1'
+  kind: Pod
+  metadata:
+    name: mypod
+    namespace: myns
+  spec:
+    containers:
+      - name: mypod
+        image: redis
+        volumeMounts:
+          - name: foo
+            mountPath: /etc/foo
+            readOnly: true
+    volumes:
+      - name: foo
+        secret:
+          secretName: mysecret
+```
+
+In this scenario, the user can specify the path where the secret is mounted. Also, notice that *myscret* is a dictionary containing multiple keys. The path to the *username* value, for example, would be `/etc/foo/username`. This is a bit different, and more complex, than Docker's approach.
+
+It's also possible for the user to modify the path where secrets are mounted. Consider the following:
+
+```yaml
+apiVersion: 'v1'
+  kind: Pod
+  metadata:
+    name: mypod
+    namespace: myns
+  spec:
+    containers:
+      - name: mypod
+        image: redis
+        volumeMounts:
+          - name: foo
+            mountPath: /etc/foo
+            readOnly: true
+    volumes:
+      - name: foo
+        secret:
+          secretName: mysecret
+          items:
+            - key: username
+              path: my-group/my-username
+```
+
+By adding `items` to the `volumes` definition above, the path to access the *username* is `/etc/foo/my-group/my-username`, and the password is no longer available, because it was not enumerated in `items`.
+
+Alternatively, secrets can be made available to a pod as an environment variables. Here's an example:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: secret-env-pod
+spec:
+  containers:
+    - name: mycontainer
+      image: redis
+      env:
+        - name: SECRET_USERNAME
+          valueFrom:
+            secretKeyRef:
+              name: mysecret
+              key: username
+        - name: SECRET_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mysecret
+              key: password
+  restartPolicy: Never
+ ```
+
+## Secrets in container.yml
+
+The following describes an approach to representing secrets in `container.yml`. It attempts to cover all of the options and variations required by each engine.
+
+### Service level secrets
+ 
+At the service level, we need the ability to specify engine specific attributes that control how a secret manifests in a container. For example, in the case of Docker there is a short form and long form for specifying a secret. In the case of K8s, a secret can be mounted as a volume, or it can be surfaced as an environment variable, and each option brings a set of attributes that users will want to set. 
+
+The following shows an approach that provides maximum flexibility at the service level:
+
+Here's the vault file:
+
+```yaml
+# vault.yml
+---
+web_password: mysql
+```
+
+And, here's the `container.yml`: 
+
+```yaml
+version: '2'
+settings:
+  vault_files:
+    - vault.yml
+  vault_password_file: /etc/vault/password.txt
+
+services:
+  web:
+    command: []
+    ports: []
+    
+    secrets:
```

Comment:
One aspect I worry about, and this is part of the Docker implementation I know less about, is whether the top level sections are passed directly to docker-compose or follow the docker-compose model. Since `secrets` is a top level version 3 docker-compose entity I was unclear if customizing it would clash or if we have more control of that?